src/backend/downloadmanager/downloadqueue.ts

```diff
@@ -0,0 +1,160 @@
+import { logError, logInfo, LogPrefix } from '../logger/logger'
+import { getMainWindow } from '../utils'
+import Store from 'electron-store'
+import { DMQueueElement } from 'common/types'
+import { installQueueElement } from './utils'
+
+const downloadManager = new Store({
+  cwd: 'store',
+  name: 'download-manager'
+})
+
+/* 
+#### Private ####
+*/
+
+type DownloadManagerState = 'idle' | 'running'
+let queueState: DownloadManagerState = 'idle'
+
+function getFirstQueueElement() {
+  if (downloadManager.has('queue')) {
+    const elements = downloadManager.get('queue') as DMQueueElement[]
+    return elements.at(0)
+  }
+
+  return null
+}
+
+function addToFinished(element: DMQueueElement) {
+  if (!element) {
+    logError('Can not add undefined element to finished!', {
+      prefix: LogPrefix.DownloadManager
+    })
+    return
+  }
+
+  let elements: DMQueueElement[] = []
+  if (downloadManager.has('finished')) {
+    elements = downloadManager.get('finished') as DMQueueElement[]
+  }
+
+  const elementIndex = elements.findIndex(
```

Comment:
I wonder if this could ever happen, or if this happened, why? I can imagine installing a game, uninstalling it, and installing it again, in that case I think it's ok to show the game twice maybe?

I asking just to simplify this, if this shouldn't happen maybe we don't need to add this complexity and we can simply push elements when done